### PR TITLE
build: update go modules

### PR DESCRIPTION
**Pull request by BlaineEXE**
Update go modules. The primary goal is to remove the goproxy import which has security warnings. The library is not used directly or indirectly by any called Rook code, only dependency modules. Therefore, exclude it from the build.

At the same time, portworx deps are starting to complicate the module update again. Rook also doesn't directly or indirectly use the portworx code, so exclude it also.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -110,50 +111,49 @@ require (
...
+	sigs.k8s.io/kustomize/kyaml v0.14.3 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.3.0 // indirect
 )
 
-replace (
-	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
-	github.com/kubernetes-incubator/external-storage => github.com/libopenstorage/external-storage v0.20.4-openstorage-rc3
-	github.com/portworx/sched-ops => github.com/portworx/sched-ops v0.20.4-openstorage-rc3
-)
+replace github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
```
Comment (BlaineEXE) [inline go.mod:148]: I tried updating this as well with no success. The latest is github.com/**google**/gnostic v0.6.9.

**Review by subhamkrai — COMMENTED** [review]
I think the build issues could be resolved by applying anyone of the last two comment https://github.com/kubernetes/client-go/issues/1084

**Comment by subhamkrai** [comment]
> I think the build issues could be resolved by applying anyone of the last two comment [kubernetes/client-go#1084](https://github.com/kubernetes/client-go/issues/1084)

#12672 will probably fix this issue

**Comment by subhamkrai** [comment]
@BlaineEXE ready to rebase?

**Pull request merged**